### PR TITLE
PUBDEV-7461: Add information about the h2o.get_leaderboard function to the AutoML User Guide page

### PR DESCRIPTION
**Pull request by hannah-tillman**
https://0xdata.atlassian.net/browse/PUBDEV-7461

### CODE REVIEW

**Comment by angela0xdata** [comment]
@ledell @hannah-tillman I reworked the AutoML topic slightly to (hopefully) give it a more holistic approach. So this now has the AutoML interface section with an example, a simple Predict section with a continuation of that example, and the AutoML Output section with still another continuation of that example (including showing more about the `extra_columns` parameter).

File: h2o-docs/src/product/automl.rst [hunk 1]
```diff
@@ -279,31 +264,77 @@ Here’s an example showing basic usage of the ``h2o.automl()`` function in *R*
...
+Code Examples
+~~~~~~~~~~~~~
+
+Using the previous example, you can retrieve the leaderboard as follows:
+
+.. tabs::
+   .. code-tab:: r R
+
+        # Get leaderboard with 'extra_columns = 'ALL'
+        lb_all <- h2o.get_leaderboard(object = aml, extra_columns = 'ALL')
```
Comment (ledell) [inline h2o-docs/src/product/automl.rst:326]:
```suggestion
        lb <- h2o.get_leaderboard(object = aml, extra_columns = 'ALL')
```

File: h2o-docs/src/product/automl.rst [hunk 2]
```diff
@@ -279,31 +264,77 @@ Here’s an example showing basic usage of the ``h2o.automl()`` function in *R*
...
+        lb_all
```
Comment (ledell) [inline h2o-docs/src/product/automl.rst:327]:
```suggestion
        lb
```

File: h2o-docs/src/product/automl.rst [hunk 3]
```diff
@@ -279,31 +264,77 @@ Here’s an example showing basic usage of the ``h2o.automl()`` function in *R*
...
+
+   .. code-tab:: python
+
+        # Get leaderboard with `extra_columns` = 'ALL'
+        lb_all = h2o.automl.get_leaderboard(aml, extra_columns = 'ALL')
```
Comment (ledell) [inline h2o-docs/src/product/automl.rst:332]:
```suggestion
        lb = h2o.automl.get_leaderboard(aml, extra_columns = 'ALL')
```

File: h2o-docs/src/product/automl.rst [hunk 4]
```diff
@@ -279,31 +264,77 @@ Here’s an example showing basic usage of the ``h2o.automl()`` function in *R*
...
+        lb_all
```
Comment (ledell) [inline h2o-docs/src/product/automl.rst:333]:
```suggestion
        lb
```

**Pull request merged**